### PR TITLE
[MRG] Make plot_evoked be able to also plot EOG

### PR DESCRIPTION
**Pull request by wmvanvliet**
It's sometimes useful to plot EOG evokeds, for example to verify that `find_eog_onsets` did its job correctly. This PR adds `eog` to the list of valid channel types for `mne.viz.plot_evoked`.

### CODE REVIEW

File: mne/defaults.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
               eeg=(-200., 200.), misc=(-5., 5.),
               seeg=(-200., 200.), dipole=(-100., 100.), gof=(0., 1.)),
     titles=dict(eeg='EEG', grad='Gradiometers', mag='Magnetometers',
-                misc='misc', seeg='sEEG', dipole='Dipole',
+                misc='misc', seeg='sEEG', dipole='Dipole', eog='EOG',
```
Comment (agramfort) [inline mne/defaults.py]: no need to update ylim too?

Comment (wmvanvliet) [inline mne/defaults.py]: ah, yes!

Comment (wmvanvliet) [inline mne/defaults.py]: the `ylims` in the defaults file are ignored by this function. But it does make sense to have an ylim in the defaults for EOG. For example, if you create a custom layout with both EEG and EOG channels (maybe the EOG in a separate column) and use `plot_evoked_topo`, *then* you would need an `ylim` for EOG. At any rate, I cannot think of a reason to *not* have a default ylim for EOG channels.

**Comment by agramfort** [comment]
+1 for merge.

**Comment by Eric89GXL** [comment]
@wmvanvliet this means that any `Evoked` dataset that has EOG channels will now have `evoked.plot()` show another row with EOG, right?

**Comment by agramfort** [comment]
what??? hum if so I regret merging... that's a very invasive change....

File: mne/viz/evoked.py [hunk 2]
```diff
@@ -185,7 +185,7 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
     titles = _handle_default('titles', titles)
     units = _handle_default('units', units)
     # Valid data types ordered for consistency
-    channel_types = ['eeg', 'grad', 'mag', 'seeg', 'dipole', 'gof']
+    channel_types = ['eeg', 'grad', 'mag', 'eog', 'seeg', 'dipole', 'gof']
```
Comment (Eric89GXL) [inline mne/viz/evoked.py]: I think this line determines which types get auto-plotted @agramfort, it's what I needed to change to get this plotting function to auto-plot the `DipoleFixed` data

**Comment by Eric89GXL** [comment]
(and @agramfort it doesn't look like it has been merged yet)

**Comment by agramfort** [comment]
indeed I was careful to just give a +1 and wait for your +1 :)

I remove it then ;)

File: mne/viz/evoked.py (hunk 2 above)
Comment (wmvanvliet) [inline mne/viz/evoked.py]: it also determines what can be plotted **at all**... let's see if I can separate the two.

Comment (Eric89GXL) [inline mne/viz/evoked.py]: You should be able to add another kwarg like plot_types that defaults to
what channel_types used to be. You could add an 'all' convenience shortcut
option to plot all available types.

**Comment by wmvanvliet** [comment]
How do you feel about this approach? We don't show EOG by default, but if the user adds it to the `pick` parameter manually, we do plot it. Should we allow plotting EMG and ECG as well?

**Comment by Eric89GXL** [comment]
It seems less easy to use, but I forgot there was a picks argument which
would be a bit redundant. So what is the default for picks going to be,
data channels only for backward compatibility?

**Comment by agramfort** [comment]
I agree it requires you to set picks manually but you can already show the
eog in evoked.plot right?

**Comment by wmvanvliet** [comment]
> you can already show the eog in evoked.plot right?

Before this PR, you just can't. This PR makes it so you can by manually adding it to picks.

> So what is the default for picks going to be, data channels only for backward compatibility?

My proposal is to keep the behavior before this PR as the default behavior to maintain backward compatibility.

**Comment by agramfort** [comment]
LGTM

MRG?

**Comment by dengemann** [comment]
Godd call! Thanks @wmvanvliet! Also my +1.

**Comment by wmvanvliet** [comment]
so what about EMG, ECG and other types? Does anyone remember the reason for `channel_types` (renamed `valid_channel_types` in this PR) to exist in the first place?

**Comment by dengemann** [comment]
> to exist in the first place?

yes it was just a matter of controlling things and dealing with what we know and what is asked for. the idea was to grow it as we go, it seems you correctly picked that up :)

**Comment by dengemann** [comment]
And yes, why not also support EMG and ECG.

**Comment by agramfort** [comment]
LGTM

@jaeilepp works for you?

**Comment by jaeilepp** [comment]
Well, actually eog channels are excluded by default when you do ``epochs.average``. It kind of feels silly to leave it out from the plot if the user has explicitly included it with picks when averaging (``epochs.average(picks=picks)``). I'd plot all the channel types included in the evoked data set. Wdyt?

**Comment by agramfort** [comment]
agreed yet it would change the default behavior.

what do others think?

**Comment by wmvanvliet** [comment]
> @dengemann: the idea was to grow it as we go, it seems you correctly picked that up :)

> @jaeilepp: I'd plot all the channel types included in the evoked data set. 

Can I assume that is a +2 for changing the default behaviour?
I'm +0 myself, it seems such an obscure use case that it doesn't seem worth it to break default behaviour, then again it is such a small use case that a chance in this is very unlikely to break someone's code.

**Comment by Eric89GXL** [comment]
Yeah I'm fine with changing the default behavior too because, as you mention, it's an odd case. Just note it clearly in the API section

**Comment by dengemann** [comment]
yep sounds good to me, let's do it.

**Comment by wmvanvliet** [comment]
This needs some testing to see if the default scalings make sense. Does anyone have data with ECG and/or EMG channels?

File: doc/whats_new.rst [hunk 3]
```diff
@@ -303,6 +303,8 @@ API
 
     - ``RawBrainVision`` objects now always have event channel ``'STI 014'``, and recordings with no events will have this channel set to zero by `Eric Larson`_
 
+    - EOG, ECG and EMG channels are now plotted by default (if present in data) when using :func:`mne.viz.plot_evoked` by `Marijn van Vliet`_
+
```
Comment (jaeilepp) [inline doc/whats_new.rst]: Wrong version.

**Comment by dengemann** [comment]
> Does anyone have data with ECG and/or EMG channels?

yep, will look into it later.

File: mne/viz/evoked.py [hunk 4]
```diff
@@ -185,10 +185,11 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
     titles = _handle_default('titles', titles)
     units = _handle_default('units', units)
     # Valid data types ordered for consistency
-    channel_types = ['eeg', 'grad', 'mag', 'seeg', 'dipole', 'gof']
+    valid_channel_types = ['eeg', 'grad', 'mag', 'seeg', 'eog', 'ecg', 'emg',
+                           'dipole', 'gof']
 
     if picks is None:
-        picks = list(range(info['nchan']))
+        picks = range(info['nchan'])
```
Comment (agramfort) [inline mne/viz/evoked.py]: why? it's for py3k range generators

**Comment by agramfort** [comment]
just fake an ecg / emg channel if it's for testing.

**Comment by wmvanvliet** [comment]
> just fake an ecg / emg channel if it's for testing.

I need to know the proper magnitude for these kinds of signals. Is ECG measured in uV or in mV? how about EMG?

**Comment by agramfort** [comment]
I've looked into one of my datasets with ECG. ECG is between -300uV and
300uV

**Comment by agramfort** [comment]
can you rebase to make CIs happy?

thx

**Comment by wmvanvliet** [comment]
>I've looked into one of my datasets with ECG. ECG is between -300uV and
300uV

Alright, the current code should work then.

**Comment by Eric89GXL** [comment]
Thanks @wmvanvliet

**Pull request merged**